src/WaitStatus.h

```diff
@@ -8,13 +8,16 @@
 #include <memory>
 #include <vector>
 
+#include "core.h"
+
 namespace rr {
 
 class RecordTask;
 
 class WaitStatus {
 public:
-  explicit WaitStatus(int status = 0) : status(status) {}
+  explicit WaitStatus(int status = 0) : _kind(WAIT_STATUS), status(status) {}
+  explicit WaitStatus(siginfo_t infop) : _kind(WAIT_INFO), info(infop) {}
```

Comment:
Why did you choose to have two different `WaitStatus` states instead of converting the `siginfo_t` into a regular status and adding a single flag to disambiguate `stopped()`? Seems like the latter would have been simpler?

Reply:
Frankly, I was concerned I'd get it wrong and get very confused, so I thought this would help work through the more substantive issues. I can switch the implementation to what you suggest now that things are working.